src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -2257,7 +2257,12 @@ public override BoundNode VisitDynamicObjectInitializerMember(BoundDynamicObject
             return null;
         }
 
-#endregion Visitors
+        protected override void ConditionallyAssignNullCoalescingOperator(BoundNullCoalesingAssignmentOperator node)
+        {
+            Assign(node.LeftOperand, node.RightOperand);
```

Comment:
My understanding of the semantics is that htis could never happen, right? That is, in order to prove that the left is always assigned, we would need to prove that it's always null, which is never true by definite assignment rules, right?

Reply:
Talked with Andy offline. This is needed for region analysis.

---
In reply to: [199978389](https://github.com/dotnet/roslyn/pull/28253#discussion_r199978389) [](ancestors = 199978389)

Reply:
Yup. That does mean you'll need to add region analysis tests to verify this, though. :)